users-backend/users/internal_api/permissions.py

```diff
@@ -1,23 +1,39 @@
+from typing import Any
+
 import jwt
 from pythonit_toolkit.headers import SERVICE_JWT_HEADER
 from pythonit_toolkit.pastaporto.tokens import decode_service_to_service_token
-from starlette.requests import Request
+from strawberry.permission import BasePermission
+from strawberry.types import Info
+
 from users.settings import SERVICE_TO_SERVICE_SECRET
 
 
-def is_service(request: Request) -> bool:
-    token = request.headers.get(SERVICE_JWT_HEADER)
-    secret = str(SERVICE_TO_SERVICE_SECRET)
-
-    try:
-        decode_service_to_service_token(
-            token, secret, issuer="gateway", audience="users-service"
-        )
-        return True
-    except (
-        jwt.DecodeError,
-        jwt.InvalidIssuerError,
-        jwt.ExpiredSignatureError,
-        jwt.InvalidAudienceError,
-    ):
-        return False
+def IsService(allowed_callers: list[str]):
```

Comment:
@patrick91 Strawberry permissions can't accept parameters, so I am forced to do this like this, I wonder if it might be an improvement we can do to Strawberry to instead of accepting a list of classes as `permission_classes` it accepts the instances 🤔